src/recipes.jl

```diff
@@ -280,18 +280,24 @@ end
     end
     fillrange := nothing
     seriestype := :path
+    if plotattributes[:linecolor] == :auto && plotattributes[:marker_z] !== nothing && plotattributes[:line_z] == nothing
+        line_z := plotattributes[:marker_z]
+    end
 
-    # create a secondary series for the markers
+    # create a primary series for the markers
     if plotattributes[:markershape] != :none
+        tmplabel = deepcopy(plotattributes[:label])
```

Comment:
is that neccessary or would a `copy` do?

Reply:
`copy` won't do it since there is no copy method `copy(::String)`.
But a usual assignment `tmplabel = plotattributes[:label]` should be sufficient though. 
Just used `deepcopy` to be 100% sure.

Reply:
`deepcopy` is quite slow though. I think it should work without any copying of the labels, when you just switch the primary-ness, doesn't it?

Reply:
Yes you are right.
So changing the primary-ness is sufficient.